src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmParticipantSpec.scala

```diff
@@ -0,0 +1,399 @@
+          )
+      }
+
+      // the load agent should send a CompletionMessage
+      scheduler.expectMsg(CompletionMessage(2, None))
+
+    }
+
+    s"go to $SimulateGrid when it receives an activity start trigger" in {
+
+      val activityStartTriggerId = 3
+
+      // send init data to agent
+      scheduler.send(
+        gridAgentWithParticipants,
+        TriggerWithIdMessage(
+          ActivityStartTrigger(3600),
+          activityStartTriggerId,
+          gridAgentWithParticipants
+        )
+      )
+
+      // we expect a completion message
+      scheduler.expectMsgPF() {
+        case CompletionMessage(
+              triggerId,
+              Some(Vector(ScheduleTriggerMessage(triggerToBeScheduled, _)))
+            ) =>
+          triggerId shouldBe 3
+          triggerToBeScheduled shouldBe StartGridSimulationTrigger(3600)
+        case x =>
+          fail(
+            s"Invalid message received when expecting a completion message after activity start trigger. Message was $x"
+          )
+      }
+
+    }
+
+    s"check the request asset power message indirectly" in {
+
+      val startGridSimulationTriggerId = 4
+      val firstSweepNo = 0
+
+      val voltageEhv =
+        Array(
+          (
+            Quantities.getQuantity(380, KILOVOLT),
+            Quantities.getQuantity(0, KILOVOLT)
+          ),
+          (
+            Quantities.getQuantity(380, KILOVOLT),
+            Quantities.getQuantity(0, KILOVOLT)
+          )
+        )
+
+      val powerEhv =
+        Array(
+          (
+            Quantities.getQuantity(135.90837346741768, MEGAWATT),
+            Quantities.getQuantity(60.98643348675892, MEGAVAR)
+          ),
+          (
+            Quantities.getQuantity(135.90837346741768, MEGAWATT),
+            Quantities.getQuantity(60.98643348675892, MEGAVAR)
+          )
+        )
+
+      // send the start grid simulation trigger
+      // the gird agent should send a RequestAssetPowerMessage to the load agent
+      scheduler.send(
+        gridAgentWithParticipants,
+        TriggerWithIdMessage(
+          StartGridSimulationTrigger(3600),
+          startGridSimulationTriggerId,
+          gridAgentWithParticipants
+        )
+      )
+
+      // we expect a request for voltage values of our slack node
+      // (voltages are requested by our agent under test from the superior grid)
+      val firstSlackVoltageRequestSender =
+        superiorGridAgent.expectSlackVoltageRequest(firstSweepNo)
+
+      // we now answer the request of our gridAgentsWithParticipants
+      // with a fake slack voltage message
+      superiorGridAgent.gaProbe.send(
+        firstSlackVoltageRequestSender,
+        ProvideSlackVoltageMessage(
+          firstSweepNo,
+          Seq(
+            ExchangeVoltage(
+              supNodeA.getUuid,
+              voltageEhv(firstSweepNo)._1,
+              voltageEhv(firstSweepNo)._2
```

Comment:
Using `_1` and `_2` on tuples is discouraged because it hides the name and/or type of the variable retrieved. Also check the open codacy issues for this.
So maybe implement something that does not rely on tuples?